components/server/src/workspace/workspace-starter.ts

```diff
@@ -486,20 +488,27 @@ export class WorkspaceStarter {
             if (WorkspaceImageSourceDocker.is(imgsrc)) {
                 let source: WorkspaceInitializer;
                 const disp = new DisposableCollection();
-                let checkoutLocation = this.getCheckoutLocation(workspace);
+                let checkoutLocation = CommitContext.is(workspace.context) && workspace.context.checkoutLocation || '.';
                 if (!AdditionalContentContext.hasDockerConfig(workspace.context, workspace.config) && imgsrc.dockerFileSource) {
                     // TODO(se): we cannot change this initializer structure now because it is part of how baserefs are computed in image-builder.
                     // Image builds should however just use the initialization if the workspace they are running for (i.e. the one from above).
-                    const { git, disposable } = await this.createGitInitializer({ span }, workspace, {
+
+                    const { initializer, disposable } = await this.createCommitInitializer({span}, workspace, {
                         ...imgsrc.dockerFileSource,
+                        checkoutLocation: ".",
                         title: "irrelevant",
                         ref: undefined,
                     }, user);
                     disp.push(disposable);
+                    let git: GitInitializer;
+                    if (initializer instanceof CompositeInitializer) {
+                        // we use the first git initializer for image builds only
+                        git = initializer.getInitializerList()[0].getGit()!;
+                    } else {
+                        git = initializer;
+                    }
                     git.setCloneTaget(imgsrc.dockerFileSource.revision);
                     git.setTargetMode(CloneTargetMode.REMOTE_COMMIT);
-                    checkoutLocation = "."
-                    git.setCheckoutLocation(checkoutLocation);
```

Comment:
Why do we no longer set the `checkoutLocation` for image builds? :thinking: Does it default to `.` anyway?
Also: why was the side-effect here required in the first place? :scratch:

Reply:
I'm setting it in line 498